### PR TITLE
OP2 Sprint 1

### PR DESCRIPTION
**Pull request by Dextyh**

### CODE REVIEW

File: Computer_Vision/CMakeLists.txt [hunk 1]
```diff
@@ -14,13 +14,14 @@ find_package(SQLITE3 REQUIRED)
...
 add_subdirectory(gtest)
 
 
 if(WIN32)
-    target_link_libraries(Computer_Vision ${OpenCV_LIBS} sqlite3::sqlite3 project_files)
+    target_link_libraries(Computer_Vision ${OpenCV_LIBS} project_files sqlite3::sqlite3)
 else()
-    target_link_libraries(Computer_Vision ${OpenCV_LIBS} sqlite3 dl project_files)
+    target_link_libraries(Computer_Vision ${OpenCV_LIBS} project_files sqlite3 dl)
 endif()
```
Comment (leventekobor) [inline Computer_Vision/CMakeLists.txt]: missing new line at the end of a file

File: Computer_Vision/files/CMakeLists.txt [hunk 2]
```diff
@@ -1 +1 @@
-add_library(project_files initial.h initial.cpp)
\ No newline at end of file
+add_library(project_files sqlite_functions.cpp sqlite_functions.h detect_circle.cpp detect_circle.h blur.cpp blur.h)
```
Comment (leventekobor) [inline Computer_Vision/files/CMakeLists.txt]: missing a new line at the end of a file

File: Computer_Vision/files/blur.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#include "blur.h"
```
Comment (leventekobor) [inline Computer_Vision/files/blur.cpp]: I would not make a separate file for this. We are going to blur every image so this functionality has a nice place in the main. Also here we just warp the `cv` functions to our own `void` functions.

File: Computer_Vision/files/detect_circle.cpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
+#include "detect_circle.h"
+
+int detectCircle(cv::Mat &image)
+{
+    if(!image.empty()) {
```
Comment (leventekobor) [inline Computer_Vision/files/detect_circle.cpp]: why check if the image is empty here? We already check that in main

File: Computer_Vision/files/detect_circle.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+        cv::Mat gray;
+        cvtColor(image, gray, cv::COLOR_BGR2GRAY);
```
Comment (leventekobor) [inline Computer_Vision/files/detect_circle.cpp]: we could simply load the picture in greyscale in the main

File: Computer_Vision/files/detect_circle.cpp [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            cv::putText(image, "X: " + xString, cv::Point(centerX - r/2 - r/4, centerY + r/3), cv::FONT_HERSHEY_SIMPLEX, 0.5, cv::Scalar(255,0,255), 2, 8, false);
+
+            std::string yString = std::to_string(centerY);
+            cv::putText(image, "Y: " + yString, cv::Point(centerX + r/4, centerY + r/3), cv::FONT_HERSHEY_SIMPLEX, 0.5, cv::Scalar(255,0,255), 2, 8, false);
+
+            numberOfCircles++;
+        }
+        return numberOfCircles;
+    }
+    return -1;
```
Comment (leventekobor) [inline Computer_Vision/files/detect_circle.cpp]: or we could just return the `numberOfCircles` here

File: Computer_Vision/files/sqlite_functions.cpp [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    char *cSet = new char[setNewValue.length() + 1];
+    strcpy(cSet, setNewValue.c_str());
+
+    sprintf(buffer, "UPDATE %s  SET %s WHERE id = %d;",cTable, cSet, id); // table, set, condition
+
+    sqlite3_exec(db, buffer, callback, 0, &errorMsg);
+
+    return 1;
+}
+
```
Comment (leventekobor) [inline Computer_Vision/files/sqlite_functions.cpp]: why so many new lines at the end of a file?

File: Computer_Vision/files/sqlite_functions.h [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+static int callback(void *NotUsed, int argc, char **argv, char **azColName);
+int SQLgeneralConvert(const std::string &sqlCommand, sqlite3* db);
+int SQLcreateRecord(std::string tableName, std::string path, long long int processTime, int circles, sqlite3* db);
+int SQLselectQuery(std::string values, std::string table, sqlite3* db);
+int SQLdeleteById(const std::string &table, int id, sqlite3* db);
+int SQLdeleteLast(const std::string &table, sqlite3* db);
+int SQLupdateRecord(const std::string &table, std::string setNewValue, int id, sqlite3* db);
+
+
+#endif
```
Comment (leventekobor) [inline Computer_Vision/files/sqlite_functions.h]: missing a new line at the end of a file

File: Computer_Vision/gtest/unit_test.cpp [hunk 9]
```diff
@@ -1,6 +1,31 @@
...
+{
+    cv::Mat image = cv::imread("../../img/ball.jpg");
+    EXPECT_EQ(detectCircle(image), 3);
+}
+
+TEST(DetectCircles, overlayCircles)
+{
+    cv::Mat image = cv::imread("../../img/balls11.jpg");
+    EXPECT_EQ(detectCircle(image), 3);
 }
```
Comment (leventekobor) [inline Computer_Vision/gtest/unit_test.cpp]: missing a new line here as well

File: Computer_Vision/main.cpp [hunk 10]
```diff
@@ -1,9 +1,63 @@
...
+    } else {
+        fprintf(stderr, "Opened database successfully\n");
+    }
+
+    sqlite3_open("../files/opencv-2.db", &db);
+
+    std::string imagePath("../img/baldasdls11.jpg");
+    image = cv::imread(imagePath, cv::IMREAD_COLOR);
+
+    if(argc > 1) {
```
Comment (leventekobor) [inline Computer_Vision/main.cpp]: why check `argc` here?

**Review by leventekobor — CHANGES_REQUESTED** [review]
Overall great work. 
For the folder strcutre I would suggest a change:
```
files
src (here comes all the cpp and hpp files from files)
gtest
img
docs
README.md
.
.
.
```

**Review by leventekobor — COMMENTED** [review]
Also to the testing part. I know I was not specific but I require way more tests there.

File: Computer_Vision/main.cpp (hunk 10 above)
Comment (Dextyh) [reply Computer_Vision/main.cpp]: DONE

File: Computer_Vision/gtest/unit_test.cpp (hunk 9 above)
Comment (Dextyh) [reply Computer_Vision/gtest/unit_test.cpp]: DONE

File: Computer_Vision/files/sqlite_functions.h (hunk 8 above)
Comment (Dextyh) [reply Computer_Vision/files/sqlite_functions.h]: DONE

File: Computer_Vision/files/sqlite_functions.cpp (hunk 7 above)
Comment (Dextyh) [reply Computer_Vision/files/sqlite_functions.cpp]: DONE

File: Computer_Vision/files/detect_circle.cpp (hunk 6 above)
Comment (Dextyh) [reply Computer_Vision/files/detect_circle.cpp]: DONE

File: Computer_Vision/files/detect_circle.cpp (hunk 5 above)
Comment (Dextyh) [reply Computer_Vision/files/detect_circle.cpp]: DONE

File: Computer_Vision/files/detect_circle.cpp (hunk 4 above)
Comment (Dextyh) [reply Computer_Vision/files/detect_circle.cpp]: DONE

File: Computer_Vision/files/blur.cpp (hunk 3 above)
Comment (Dextyh) [reply Computer_Vision/files/blur.cpp]: DONE

File: Computer_Vision/files/CMakeLists.txt (hunk 2 above)
Comment (Dextyh) [reply Computer_Vision/files/CMakeLists.txt]: DONE

File: Computer_Vision/CMakeLists.txt (hunk 1 above)
Comment (Dextyh) [reply Computer_Vision/CMakeLists.txt]: DONE

**Review by leventekobor — APPROVED** [review]

**Pull request merged**